File: cpp/Configuration.cpp
// =================================================================
// FILE: Configuration.cpp
// =================================================================
#include "Configuration.h"

// The private constructor initializes the configuration values.
// These values are loaded once when getInstance() is first called.
Configuration::Configuration()
{
    basePrice = 1000;
    unitPrice = 1000;
    weightLimit = 10;     // 10 kg
    dimensionLimit = 100; // Example: 100 cm for sum of length and width
}

// The Meyers' Singleton implementation.
// The instance is created on first access and is thread-safe in C++11+.
Configuration &Configuration::getInstance()
{
    static Configuration instance;
    return instance;
}

int Configuration::getBasePrice() const { return basePrice; }
int Configuration::getUnitPrice() const { return unitPrice; }
int Configuration::getWeightLimit() const { return weightLimit; }
int Configuration::getDimensionLimit() const { return dimensionLimit; }

File: python/configuration.py
class Configuration:
    """A singleton class to hold global application configuration."""
    _instance = None

    @classmethod
    def get_instance(cls) -> 'Configuration':
        """Class method to get the single instance of the class."""
        if cls._instance is None:
            cls._instance = cls()
        return cls._instance

    def __init__(self):
        if Configuration._instance is not None:
            raise RuntimeError("This class is a singleton! Use get_instance().")
        else:
            self.base_price = 1000
            self.unit_price = 1000
            self.weight_limit = 10
            self.dimension_limit = 100